src/crypto/crypto_keygen.cc

```diff
@@ -67,11 +68,12 @@ Maybe<bool> SecretKeyGenTraits::AdditionalConfig(
   CHECK(args[*offset]->IsUint32());
   params->length = std::trunc(args[*offset].As<Uint32>()->Value() / CHAR_BIT);
   if (params->length > INT_MAX) {
-    char msg[1024];
-    snprintf(msg, sizeof(msg),
-             "length must be less than or equal to %lu bits",
-             static_cast<size_t>(INT_MAX) * CHAR_BIT);
-    THROW_ERR_OUT_OF_RANGE(env, msg);
+    const std::string msg {
+      "length must be less than or equal to "
+      + std::to_string(static_cast<uint64_t>(INT_MAX) * CHAR_BIT)
+      + " bits"
+    };
```

Comment:
You can also use `SPrintF()` as we do in other parts of the source :)

Reply:
@addaleax thanks for the suggestion. :)

Do you mean formatting the product as a string by first passing it through `to_string` first as I've done here?
I have tried formatting it as a number and it's causing some problems related to the architecture:

### Formatting `uint64_t`

|           | 32-bit architectures        | 64-bit architectures        |
|    ---   | ---                                   | ---                                    |
|`%lu`  | :heavy_multiplication_x: | :heavy_check_mark:      |
|`%llu` | :heavy_check_mark:      | :heavy_multiplication_x: |

Ref: https://github.com/nodejs/node/actions/runs/355453238

Reply:
> Do you mean formatting the product as a string by first passing it through `to_string` first as I've done here?

No :) Our `SPrintF()` mostly disregards the actual format string and converts everything to string: 

https://github.com/nodejs/node/blob/be1788613e28f2558b8fc26b6178f434da8ad79e/test/cctest/test_util.cc#L259

> I have tried formatting it as a number and it's causing some problems related to the architecture:

`%zu` is the portable version that we could use for integers of type `size_t` (which we could use in the current code as well in order to fix this). For `uint64_t`, you could use `PRIu64`, as in, `snprintf(..., "%" PRIu64 "\n", uint64);`.

Reply:
I don't think `size_t` is the right thing to use here as the product exceeds 32 bits and `size_t` holds 32 bits on 32-bit architectures. I'll use `PRIu64` then, thanks! :)

Reply:
@RaisinTen That’s totally fine, but if you do stick with `std::string`, I’d suggest just sticking with `SPrintF` :)

Reply:
@addaleax done! :)